lib/rubocop/cop/style/pattern_argument.rb

```diff
@@ -0,0 +1,74 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Style
+      # This cop looks for all? / any? / none? / one? calls where the block
+      # has a single statement and === has been invoked on the receiver so the
+      # === argument can be used as the predicate method argument.
+      #
+      # @example
+      #   # bad
+      #   %w[foo bar foobar].any? { |e| /oo/.match?(e) }
+      #
+      #   [1, 2, 3].any? { |e| (1..10).include?(e) }
+      #
+      #   [1, 2, 3].any? { |e| Set[1, 2, 3] === e }
+      #
+      #   [1, 2, 3].none? do |e|
+      #     Integer === e
+      #   end
+      #
+      #   # good
+      #   %w[foo bar foobar].any?(/oo/)
+      #
+      #   [1, 2, 3].any?(1..10)
+      #
+      #   [1, 2, 3].any?(Set[1, 2, 3])
+      #
+      #   [1, 2, 3].none?(Integer)
+      #
+      class PatternArgument < Cop
```

Comment:
Could you inherit from `Base` instead? See [the docs](https://docs.rubocop.org/rubocop/0.87/v1_upgrade_notes.html)

Reply:
I did that, also updated the tests to use `expect_correction`, even better. Just don't know if the autocorrect is fine now [right inside `on_block`](https://github.com/rubocop-hq/rubocop/pull/8220/files#diff-c0aa68a59ced803500d8a79422f5ee7fR44).

Reply:
Yes, that's perfect. Except you already have the `method` and `pattern`, so you should avoid calling it again in `replacement`...

Reply:
Nice, I was wondering there was something strange there 🎉